physical/file.go

```diff
@@ -99,15 +109,21 @@ func (b *FileBackend) Get(k string) (*Entry, error) {
 	b.l.Lock()
 	defer b.l.Unlock()
 
-	path, key := b.path(k)
-	path = filepath.Join(path, key)
+	basePath, fileName := b.path(k)
+	fullPath := filepath.Join(basePath, "_"+fileName)
+
+	// If non-encoded file name is a valid storage entry, read it out.
+	// Otherwise, encode the file name and look for the storage entry.
+	f, err := os.Open(fullPath)
+	if err != nil && os.IsNotExist(err) {
+		fullPath = filepath.Join(basePath, "_"+base64.URLEncoding.EncodeToString([]byte(fileName)))
```

Comment:
Rather than do the munging with base64-encoding and `_` everywhere, I'd make the `path` function return three values instead of two: the two it originally returned, and a third value with `_` plus the base64 path. This removes a lot of boilerplate elsewhere.